lib/puppet/application/agent.rb

```diff
@@ -401,8 +401,6 @@ def setup
     Puppet.settings.use :main, :agent, :ssl
 
     Puppet::Transaction::Report.indirection.terminus_class = :rest
-    # we want the last report to be persisted locally
-    Puppet::Transaction::Report.indirection.cache_class = :yaml
```

Comment:
By unsetting this, we'll use the default `cache_class`. If that's `nil` then everything is ok, but if it's `yaml` or `json` we'll end up caching the report a second time. I pretty sure that's not the case, but wanted to ask.